.github/workflows/test_examples.yml

```diff
@@ -0,0 +1,43 @@
+name: Test Examples
+on:
+  pull_request:  # Run on all PR changes
+    types: [opened, synchronize, reopened]
+  push:  # Run on all changes to main (includes PR merges)
+    branches:
+      - main
+
+jobs:
+  test-examples:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read  # Minimal permissions required for checkout
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          # We need the full history to generate the proper version number
+          fetch-depth: 0
```

Comment:
We don't care about the proper version number here, so I think we can remove this.